pymbolic/mapper/__init__.py

```diff
@@ -434,22 +434,12 @@ def map_lookup(self, expr, *args, **kwargs):
         return type(expr)(aggregate, expr.name)
 
     def map_sum(self, expr, *args, **kwargs):
-        children = [self.rec(child, *args, **kwargs) for child in expr.children]
-        if all(child is orig_child
-                for child, orig_child in zip(children, expr.children)):
-            return expr
```

Comment:
How come these got removed? IIRC @isuruf found them essential for making no-op traversals cheap in loopy.

Reply:
Yep, I'll revert them.